nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,320 @@
+        a smooth parameter value in mm used to smooth an image before locating
+        the peak local max value.
+
+    standardize: boolean, True or False, default False, optional
+        If True, the time series signals are centered and normalized by
+        putting their mean to 0 and variance to 1. Recommended to
+        set as True if signals are not already standardized.
+        passed to class NiftiMapsMasker.
+
+    detrend: boolean, True or False, default False, optional
+        This parameter is passed to nilearn.signal.clean basically
+        indicates whether to detrend timeseries signals or not.
+        passed to class NiftiMapsMasker.
+
+    low_pass: float, default is None, optional
+        This value will be applied on the signals by passing to signal.clean
+        Please see the related documentation signal.clean for more details.
+        passed to class NiftiMapsMasker.
+
+    high_pass: float, default is None, optional
+        This value will be applied on the signals by passing to signal.clean
+        Please see the related documentation signal.clean for more details.
+        passed to NiftiMapsMasker.
+
+    t_r: float, default is None, optional
+        Repetition time in sec. This value is given to signal.clean
+        Please see the related documentation for details.
+        passed to NiftiMapsMasker.
+
+    memory: instance of joblib.Memory, string, default is None, optional
+        Used to cache the masking process. If a string is given, the path
+        is set with this string as a folder name in the directory.
+        passed to NiftiMapsMasker.
+
+    memory_level: int, default is 0, optional
+        Aggressiveness of memory catching. The higher the number, the higher
+        the number of functions that will be cached. Zero mean no caching.
+        passed to NiftiMapsMasker.
+
+    verbose: int, default is 0, optional
+        Indicates the level of verbosity by printing the message. Zero indicates
+        nothing is printed.
+
+    Attributes
+    ----------
+    regions_: Niimg-like image/object
+        a list of seperated regions with each region lying on a 3D volume
+        concatenated into a 4D Nifti like object.
+
+    index_: a numpy array
+        an array of list of indices where each index value is assigned to
+        each seperate region of its corresponding family of brain maps.
+
+    signals_: a numpy array
+        a list of averaged timeseries signals of the subjects extracted from
+        each region.
+
+    References
+    ----------
+    * Abraham et al. "Region segmentation for sparse decompositions: better
+      brain parcellations from rest fMRI", Sparsity Techniques in Medical Imaging,
+      Sep 2014, Boston, United States. pp.8
+    """
+    def __init__(self, maps_img, mask_img=None, min_size=50,
+                 threshold="80%", thresholding_strategy='percentile',
+                 extractor='local_regions', peak_local_smooth=6.,
```

Comment:
peak_local_smooth doesn't strike me as a great name.

@AlexandreAbraham just curious, does it make sense to have peek_local_smooth != smoothing_fwhm in general?